docs/overview/reference/glossary.md

```diff
@@ -58,6 +58,10 @@ Mostly offline devices refer to clients who spend most of their time offline or
 
 A node is a device or client that implements Waku [protocols](#protocol) and leverages the [Waku Network](#waku) to enable secure and private peer-to-peer Web3 communication.
 
+### Node Key
+
+A node key is a [Secp256k1](https://en.bitcoin.it/wiki/Secp256k1) (64 char hex string) private key for generating a Waku node public [listening](#transport) and [discovery](#peer-discovery) address.
```

Comment:
It used to generate the Peer id of the node. Which in turns, is present in the multiaddr of the node that is used by others to connect to said node.

Finally, this is a security measure as authentication method is used by libp2p at connection time to verify that a node is truly who they are based on their peer id.